src/meetingreadinesschecker/DefaultMeetingReadinessChecker.ts

```diff
@@ -0,0 +1,451 @@
+    }
+    audioVideo.realtimeSubscribeToAttendeeIdPresence(attendeePresenceHandler);
+    if (!(await this.startMeeting())) {
+      audioVideo.realtimeUnsubscribeToAttendeeIdPresence(attendeePresenceHandler);
+      return CheckAudioConnectivityFeedback.ConnectionFailure;
+    }
+    await this.executeTimeoutTask(async () => {
+      return audioPresence;
+    });
+    audioVideo.realtimeUnsubscribeToAttendeeIdPresence(attendeePresenceHandler);
+    await this.stopMeeting();
+    return audioPresence
+      ? CheckAudioConnectivityFeedback.Success
+      : CheckAudioConnectivityFeedback.NoAudioPresence;
+  }
+
+  async checkVideoConnectivity(
+    videoInputDeviceInfo: MediaDeviceInfo
+  ): Promise<CheckVideoConnectivityFeedback> {
+    const audioVideo = this.meetingSession.audioVideo;
+
+    try {
+      const permissionResult = await audioVideo.chooseVideoInputDevice(videoInputDeviceInfo);
+      if (
+        permissionResult === DevicePermission.PermissionDeniedByBrowser ||
+        permissionResult === DevicePermission.PermissionDeniedByUser
+      ) {
+        return CheckVideoConnectivityFeedback.VideoInputPermissionDenied;
+      }
+    } catch (error) {
+      this.logger.error(
+        `MeetingReadinessChecker: Failed to get video input device with error ${error}`
+      );
+      return CheckVideoConnectivityFeedback.FailureToGetVideoInput;
+    }
+
+    if (!(await this.startMeeting())) {
+      return CheckVideoConnectivityFeedback.ConnectionFailure;
+    }
+
+    let packetsSent = 0;
+    audioVideo.startLocalVideoTile();
+    await this.executeTimeoutTask(async () => {
+      const rawStats = await audioVideo.getRTCPeerConnectionStats();
+      if (rawStats) {
+        rawStats.forEach(report => {
+          if (report.type === 'outbound-rtp' && report.mediaType === 'video') {
+            packetsSent = report.packetsSent;
+          }
+        });
+      }
+      return packetsSent > 0;
+    });
+    audioVideo.stopLocalVideoTile();
+    await this.stopMeeting();
+    if (packetsSent <= 0) {
+      return CheckVideoConnectivityFeedback.NoPackageSentToServer;
+    }
+    return CheckVideoConnectivityFeedback.Success;
+  }
+
+  async checkNetworkUDPConnectivity(): Promise<CheckNetworkUDPConnectivityFeedback> {
+    try {
+      this.originalURLRewriter = this.meetingSession.configuration.urls.urlRewriter;
+    } catch (e) {
+      this.logger.error(`Meeting session not initialized`);
+      return CheckNetworkUDPConnectivityFeedback.MeetingSessionNotInitialized;
+    }
+    this.meetingSession.configuration.urls.urlRewriter = (uri: string) => {
+      const transformedUri = this.originalURLRewriter(uri);
+      if (transformedUri.includes('transport=tcp')) {
+        return '';
+      }
+      return transformedUri;
+    };
+
+    const audioVideo = this.meetingSession.audioVideo;
+    if (!(await this.startMeeting())) {
+      this.meetingSession.configuration.urls.urlRewriter = this.originalURLRewriter;
```

Comment:
In some checker APIs, you clean up the resource in `finally` block while some APIs clean up before returning. In the next improvement PR, you can follow the `finally` pattern in case some API calls fail.

Reply:
So the reason I did that is because I am not sure if we should call audioVideo.stop if .start() failed for example.

Reply:
It would be clear if we can make `stopMeeting()` always valid for cleaning up dangling resources from `startMeeting`. Then, we can call `stopMeeting()` in `finally` block.